hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/DataEntity.java

```diff
@@ -125,21 +130,33 @@ public Blueprint build() {
     }
 
     public static final class Update extends Entity.Update {
+
+        private DataEntity.Role role;
         private final StructuredData value;
 
         public static Builder builder() {
             return new Builder();
         }
 
-        public Update(StructuredData value, Map<String, Object> properties) {
+        public Update(DataEntity.Role role, StructuredData value, Map<String, Object> properties) {
             super(properties);
+            this.role = role;
             this.value = value;
         }
 
+        // this is needed for Jackson deserialization
+        public Update() {
+            this(null, null, null);
+        }
+
         public StructuredData getValue() {
             return value;
         }
 
+        public Role getRole() {
```

Comment:
This would imply that one can change the role of the data entity. That is not the case though, so this field should not be needed.